### PR TITLE
Assert for vm_segment_destroy_range and simplyfing do_munmap

### PR DESCRIPTION
**Pull request by wiklam**
Asserting that we are destroying the range between a segment can prevent us from some future bugs.
The max for `do_munmap` is unnecessary, cause `vm_map_find_segment` already guarantee that `addr >= start`.

### CODE REVIEW

**Review by pj1031999 — APPROVED** [review]
LGTM

File: sys/kern/vm_map.c [hunk 1]
```diff
@@ -164,7 +164,8 @@ void vm_segment_destroy(vm_map_t *map, vm_segment_t *seg) {
 void vm_segment_destroy_range(vm_map_t *map, vm_segment_t *seg, vaddr_t start,
                               vaddr_t end) {
   assert(mtx_owned(&map->mtx));
-  assert(start >= vm_map_start(map) && end <= vm_map_end(map));
+  assert(seg->start >= vm_map_start(map) && seg->end <= vm_map_end(map));
```
Comment (j-piecuch) [inline sys/kern/vm_map.c:167]: Is this assertion needed? Can't we assume that all segments in the `vm_map` are valid (i.e. they fall within the boundaries of the `vm_map`)?

Comment (wiklam) [reply sys/kern/vm_map.c]: It is pretty obvious, idk why I had some doubts about it.

**Review by j-piecuch — APPROVED** [review]
LGTM. It looks like the semantics of `do_munmap()` are way too strict compared to [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/munmap.html), but I will fix that in another PR.

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**